### PR TITLE
fix: Do not overwrite button styles in tooltip lists

### LINKED ISSUE #4303: Text for the 'Log in to flag review' button placed in the Flag review section needs to be white
Follow up from [#2517](https://github.com/mozilla/addons-frontend/issues/2517)
STR:

1. Access AMO while logged out
2. Choose an add-on that has some reviews to view its details page
3. Go the add-on's reviews page and click on 'Flag' review for any of the listed reviews
4. Observe the text color of the 'Log in to flag review' button

Actual result:
Text color is black

Expected result:
Text color should be white

Reproduced on AMO dev FF58, (Win10, Android 7.0)

image

### PR DESCRIPTION
**Pull request by tofumatt**
Fixes #4303, by not overriding button styles inside these elements. The nesting on a lot of this CSS is pretty rough/complex by necessity and this was the most future-proof way to solve it for now. It could use future cleanup.

### Before
screenshot 2018-02-05 12 42 43
screenshot 2018-02-05 12 42 39

### After
screenshot 2018-02-05 12 39 02
screenshot 2018-02-05 12 38 58

### CODE REVIEW

File: src/amo/components/AddonReviewListItem/styles.scss [hunk 1]
```diff
@@ -36,9 +36,11 @@
   a:visited,
   a:hover,
   a:active {
-    color: $blue-50;
-    font-weight: normal;
-    text-decoration: none;
+    &:not(.Button) {
```
Comment (kumar303) [inline src/amo/components/AddonReviewListItem/styles.scss]: I don't see `button` in the selector list. Why was this needed?

File: src/ui/components/TooltipMenu/styles.scss [hunk 2]
```diff
@@ -70,32 +70,39 @@
...
+        padding: 0;
+        text-decoration: none;
+      }
     }
 
     a:hover,
     a:focus,
     a:active {
-      color: $blue-50;
+      &:not(.Button) {
```
Comment (kumar303) [inline src/ui/components/TooltipMenu/styles.scss]: Same question: I didn't see `button` in the selector list

File: src/amo/components/AddonReviewListItem/styles.scss (hunk 1 above)
Comment (tofumatt) [reply src/amo/components/AddonReviewListItem/styles.scss]: The buttons are `a` tags here (and in the other place) because they have a `href` attribute.

Comment (kumar303) [reply src/amo/components/AddonReviewListItem/styles.scss]: Ah, the login button is an anchor. I didn't think of that.
